sdk/search/Azure.Search.Documents/src/Models/SearchField.cs

```diff
@@ -1,12 +1,137 @@
 // Copyright (c) Microsoft Corporation. All rights reserved.
 // Licensed under the MIT License.
 
+using System;
+using System.Collections.Generic;
 using Azure.Core;
 
 namespace Azure.Search.Documents.Models
 {
+    /// <summary>
+    /// Represents a field in an index definition, which describes the name, data type, and search behavior of a field.
+    /// <para>
+    /// When creating an index, instead use the <see cref="SimpleField"/> and <see cref="ComplexField"/> classes to help you more easily create a <see cref="SearchIndex"/>.
+    /// </para>
+    /// </summary>
     [CodeGenSchema("Field")]
     public partial class SearchField
     {
+        // TODO: Replace constructor and read-only properties when https://github.com/Azure/autorest.csharp/issues/554 is fixed.
+
+        /// <summary>
+        /// Initializes a new instance of the <see cref="SearchField"/> class.
+        /// </summary>
+        /// <param name="name">The name of the field, which must be unique within the index or parent field.</param>
+        /// <param name="type">The data type of the field.</param>
+        /// <exception cref="ArgumentException"><paramref name="name"/> is an empty string.</exception>
+        /// <exception cref="ArgumentNullException"><paramref name="name"/> is null.</exception>
+        public SearchField(string name, DataType type)
+        {
+            Argument.AssertNotNullOrEmpty(name, nameof(name));
+
+            Name = name;
+            Type = type;
+        }
+
+        // TODO: Remove when https://github.com/Azure/autorest.csharp/issues/582 is fixed.
+        private SearchField()
+        {
+        }
+
+        /// <summary>
+        /// Gets the name of the field.
+        /// </summary>
+        [CodeGenSchemaMember("name")]
+        public string Name { get; }
+
+        /// <summary>
+        /// Ge the data type of the field.
+        /// </summary>
+        [CodeGenSchemaMember("type")]
+        public DataType Type { get; }
+
+        // TODO: Remove "overrides" for boolean properties when https://github.com/Azure/autorest.csharp/issues/558 is fixed.
+
+        /// <summary>
+        /// Gets or sets a value indicating whether the field is full-text searchable. The default is null.
```

Comment:
Can we add a bit more explanation of what null means?  My mental model is that it's a tri-state where null is partially searchable, but that my be incorrect.

Reply:
The only text I can think of reinforce that using this class puts everything under your control and is not advised (that text exists at the class level already). Null simply means the field isn't serialized to the server. Depending on the `type` field, that can err or use the defaults. It wouldn't been easier if the service used discriminated types here as they did elsewhere. Instead, we have this monolithic conditionally serialized type. 😞

Reply:
I think that's fair.  Maybe we can put a blurb in the `<remarks>` section about null values not being sent and being interpreted according to the type.  I don't feel super strongly - I think you're already guiding them to the better types.

Reply:
Actually, this is already done. It's the next line down. The compiler merges all text together according to XML rules, so this appears as a single sentence like,
> Gets or sets a value indicating whether the field is full-text searchable. The default is null. This means it will undergo analysis such as word-breaking during indexing. This property can be true only for <see cref="DataType.String"/> or "Collection(DataTime.String)". It must be false for non-string simple fields, and null for complex fields.

I will repeat the recommendation from the class remarks, though, in each member's remarks.